README.md

```diff
@@ -42,13 +42,32 @@ Additionally, steps must have material and product rules which define the files
 
 *Hint: Take a look at [`create_layout.py`](https://github.com/in-toto/in-toto/blob/develop/demo/owner_alice/create_layout.py), a script that creates the in-toto demo layout.*
 
-#### Rules
-*TODO*
+#### Artifact Rules
+A software supply chain usually operates on a set of files, such as source code, executables, packages, or the like. in-toto calls these files artifacts or more precisely materials and products.
+A material describes the state of an artifact before a step or inspection has been carried out and a product describes its state thereafter.
+
+The in-toto layout provides a simple rule language to authorize or enforce the artifacts that should be used as materials for or result from the execution of a step or inspection and to link subsequent steps and inspections together. This adds the following guarantees for any given step or inspection:
+- Only artifacts **authorized** by the project owner are created, modified or deleted,
+- each defined creation, modification or deletion is **enforced**, and also
+- restricted to the scope of its definition, which **links** subsequent steps and inspections together.
+
+Note that it is up to you to provide the desired security guarantees, by authorizing, enforcing and linking material and product together using any and usually multiple of the following rules:
+- `CREATE <pattern>`
+- `DELETE <pattern>`
+- `MODIFY <pattern>`
+- `ALLOW <pattern>`
+- `REQUIRE <pattern>`
+- `DISALLOW <pattern>`
+- `MATCH <pattern> [IN <source-path-prefix>] WITH (MATERIALS|PRODUCTS) [IN <destination-path-prefix>] FROM <step>`
+
+*Rule arguments specified as `<pattern>` allow for Unix shell-style wildcards as implemented by Python's [`fnmatch`](https://docs.python.org/2/library/fnmatch.html).*
+
+To learn more about the different rule types, their guarantees and how they are applied take a look at the [Artifact Rules](https://docs.google.com/document/d/1TpcMIMEKWvgRHYmFZ7OtM_iNrhmUaMUoot3XO29uhuo/edit#heading=h.fv3n0caa9km) section of the in-toto sepcification.
 
 #### Carrying out software supply chain steps
 
 ##### in-toto-run
-`in-toto-run` executes the passed command and records the path and hash of the passed *materials* - files before command execution - and the *products* - files after command execution and optionally stores them together with the *byproducts* of the command - return value, stdout, stderr - to a link file (`<step-name>.link`), signed with the functionary's key.
+`in-toto-run` executes the passed command and records the path and hash of the passed *materials* - files before command execution - and *products* - files after command execution and optionally stores them together with the *byproducts* of the command - return value, stdout, stderr - to a link file (`<step-name>.link`), signed with the functionary's key.
```

Comment:
Instead of the dashes on `return value, stdout, and stderr`, I would use parentheses and e.g.,. There is the possibility that, in the future, we gather more byproducts.